internal/providers/terraform/hcl_provider.go

```diff
@@ -600,6 +606,28 @@ func (p *HCLProvider) countReferences(block *hcl.Block) *countExpression {
 	return nil
 }
 
+var ignoredAttrs = map[string]bool{"arn": true, "id": true, "name": true, "self_link": true}
+var checksumMarshaller = jsoniter.ConfigCompatibleWithStandardLibrary
+
+func generateChecksum(value map[string]interface{}) string {
+	filtered := make(map[string]interface{})
+	for k, v := range value {
+		if !ignoredAttrs[k] {
+			filtered[k] = v
+		}
+	}
+
+	serialized, err := checksumMarshaller.Marshal(filtered)
+	if err != nil {
+		return ""
+	}
+
+	h := sha256.New()
+	h.Write(serialized)
+
+	return hex.EncodeToString(h.Sum(nil))
+}
+
```

Comment:
@hugorut Could you please review this change? I've updated it from `sha1` to `sha256` to resolve a gosec warning. `// #nosec` didn't work for me. What is the right way to deal with this?

Reply:
LGTM